### ISSUE TITLE
TypeError: '<' not supported between instances of 'int' and 'str'

### ISSUE THREAD

**Issue by pokbb** [beaston02/MFCRecorder#34]
Hello, a few days ago I've been receiving this error. It happens with some models, but when it does it usually blocks the script, it keeps recording the models that it is recording at that moment, but then it does not keep checking the page.
Thanks for your work ;)

Sorry my English.

recording FancyV: 21523305 (137 viewers) [set()]
Traceback (most recent call last):
  File "mfcrecorder.py", line 31, in <module>
    if not config.does_model_pass_filter(model):
  File "/home/pi/script/MFCRecorder/classes/config.py", line 127, in does_model_pass_filter
    if model.session['rc'] < max(m_settings['min_viewers'], m_settings['stop_viewers']):
TypeError: '<' not supported between instances of 'int' and 'str'

My config.conf configuration.

[settings]
check_interval = 30
min_space = 1
post_processing_command =
post_processing_thread_count = 1
min_viewers = 0
stop_viewers = 0
priority = 0

[auto_recording]
viewers = 0
auto_stop_viewers = 0
score = 0
newer_than_hours = 0
tags = 0
min_tags = 1
tag_viewers = 0
tag_stop_viewers = 0

**Comment by ebangbang** [comment]
I'm getting something similar (_'>' not supported between instances of 'str' and 'int'_) it keeps giving me an error for each "run" for a while and then it stops (might be related to script below)


.
```
[2017-11-14 21:11:14,665] Unexpected exception: <class 'TypeError'>
Connecting to MyFreeCams chat server xchat73...
Login handshake completed. Logged in as 'Guest12350' with sessionId 322426111_received
    self.client.packet_received(Packet(*unpacked_data[1:-1], smessage))
  File "/usr/lib/python3.6/site-packages/mfcauto/client.py", line 95, in packet_received
    self._process_packet(packet)
Connecting to MyFreeCams chat server xchat81...o/client.py", line 151, in _process_packet
Login handshake completed. Logged in as 'Guest03443' with sessionId 322427204
  File "/usr/lib/python3.6/site-packages/mfcauto/client.py", line 246, in _handle_extdata
    self._process_packet(packet)
  File "/usr/lib/python3.6/site-packages/mfcauto/client.py", line 165, in _process_packet
    self._process_packet(Packet(FCTYPE.METRICS, packet.nfrom, FCTYPE.SESSIONSTATE, len(rdata), len(rdata)))
  File "/usr/lib/python3.6/site-packages/mfcauto/client.py", line 148, in _process_packet
    self.emit(FCTYPE.CLIENT_MODELSLOADED)
  File "/usr/lib/python3.6/site-packages/mfcauto/event_emitter.py", line 24, in emit
    func(*args)
  File "/cygdrive/e/Download/Ord/qar/MFCRecorder2/classes/models.py", line 19, in on_tags
    models = {int(model.uid): Model(model) for model in all_results
  File "/cygdrive/e/Download/Ord/qar/MFCRecorder2/classes/models.py", line 20, in <dictcomp>
    if model.uid > 0 and model.bestsession['vs'] == mfcauto.STATE.FreeChat
TypeError: '>' not supported between instances of 'str' and 'int'
finished run
```

Maybe a should mention that I've made some small adjustments that could be explain why my error is a bit different.

- Got an error on "mfcauto.FCTYPE.CLIENT_TAGSLOADED" so changed that to CLIENT_MODELSLOADED (don't care about tags anyhow)

- I'm running it through Cygwin

- Some additional prints to a file instead of the console

- Also I'm running it through a script that tries to restart it automatically if the log appears to not be changing (compares last 4 lines of log every 30 sec, i'll [attach it](https://github.com/beaston02/MFCRecorder/files/1473467/keepalive_mfcrecorder.zip) if anyone is curious)

**Comment by sKanoodle** [comment]
@pokbb @ebangbang since it looks like this issue is related to mfcauto, could you both tell me, which version of mfcauto you are using?

**Comment by pokbb** [comment]
@sKanoodle 
I use the last version.
pip3.6 list
mfcauto (1.0.0)

@ebangbang 
I think i got that error one time in the last month, but I'm not sure.

**Comment by ebangbang** [comment]
Just installed in Cygwin because it stopped working on Windows, so seems my version is a bit newer, mfcauto (0.1.2).

**Comment by sKanoodle** [comment]
@ebangbang 0.1.2 is an old version, latest is 1.0.0. On 1.0.0 you will not have the problems with the FCTYPE.CLIENT_TAGSLOADED anymore.

@pokbb how often would you say do you encounter this error? I can't seem to be able to reproduce it.

**Comment by pokbb** [comment]
@sKanoodle 
Every day since 2 weeks ago or more. I`m going to test in to a virtual machine and i'm going to update the script. Thanks :)

**Comment by pokbb** [comment]
@sKanoodle
I tested the old version of MFCRecorder on ubuntu 16.04, i used same config.conf and wanted.json. I got the same error.
Later, i tested the new version of MFCRecorder and work fine (2 days). The script show the error but continue running. Example:

started recording Lady1
recording Lady1.... pr0n tags...
'<' not supported between instances of 'int' and 'str'
started recording Lady2
recording Lady2.... pr0n tags...
started recording Lady3
recording Lady3.... pr0n tags...

The issue persist but now MFCRecorder don't block. So, i think the error is solved. Thanks :)
